testing/tests/joystick.lua

```diff
@@ -0,0 +1,81 @@
+-- love.joystick
+-- @NOTE we can't test this module fully as it's hardware dependent
+-- however we can test methods do what is expected and can handle certain params
+
+--------------------------------------------------------------------------------
+--------------------------------------------------------------------------------
+------------------------------------METHODS-------------------------------------
+--------------------------------------------------------------------------------
+--------------------------------------------------------------------------------
+
+
+-- love.joystick.getGamepadMappingString
+love.test.joystick.getGamepadMappingString = function(test)
+  local mapping = love.joystick.getGamepadMappingString('faker')
+  test:assertEquals(nil, mapping, 'check no mapping for fake gui')
+end
+
+
+-- love.joystick.getJoystickCount
+love.test.joystick.getJoystickCount = function(test)
+  local count = love.joystick.getJoystickCount()
+  test:assertEquals(0, count, 'check no joysticks')
+end
+
+
+-- love.joystick.getJoysticks
+love.test.joystick.getJoysticks = function(test)
+  local joysticks = love.joystick.getJoysticks()
+  test:assertEquals(0, #joysticks, 'check no joysticks')
```

Comment:
I don't think we should check the count here, since I'd expect tests to pass on my computer with an xbox controller connected and with it not connected.

Reply:
Maybe change it to assertGreaterEqual like the getJoystickCount test above?